presentation/src/main/java/com/architect/coders/mu8/comics/ComicsActivity.kt

```diff
@@ -51,35 +42,45 @@ class ComicsActivity : AppCompatActivity() {
             )
         }
 
+        binding.viewmodel = viewModel
+        binding.lifecycleOwner = this
+
+        configureToolbar()
+
+        configureRecycler()
+
+        observerViewModel()
+    }
+
+    private fun configureToolbar() {
+        toolbar = findViewById(R.id.comic_activity_toolbar)
+        setSupportActionBar(toolbar)
+        supportActionBar?.setDisplayShowTitleEnabled(false)
+
+        toolbarTitle = findViewById(R.id.toolbar_title)
+        toolbarTitle.text = getString(R.string.comics_name)
+    }
+
+    private fun configureRecycler() {
         adapter = ComicsAdapter(viewModel::onComicClicked)
         recycler.layoutManager = GridLayoutManager(this, GRID_COLUMS)
         recycler.adapter = adapter
+    }
 
-        viewModel.model.observe(this, Observer(::updateUi))
+    private fun observerViewModel() {
+        viewModel.comics.observe(this, Observer(::showComics))
+        viewModel.messageError.observe(this, Observer(::showMessages))
+        viewModel.navegateTo.observe(this, EventObserver { comic ->
+            toast(comic.title)
```

Comment:
Can you put this and the closing curly braces/parenthesis on the same line?

Reply:
Done

Reply:
please upload the changes

Reply:
Done